packages/metrics/test/metrics.test.js

```diff
@@ -0,0 +1,193 @@
+        .then(() => {
+          expect(getMetrics()[`${resource1}:get`].requests).toBe(1);
+          expect(getMetrics()[`${resource2}:get`].requests).toBe(1);
+        });
+  });
+
+  it('doesn\'t set metrics on ignored endpoint', () => {
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => {});
+
+    runMetrics({ handler: stellar.handler }, service, `${resource2}:get`);
+
+    return stellar.request.get(resource1)
+        .then(() => {
+          expect(getMetrics()[`${resource1}:get`]).not.toBeDefined();
+        });
+  });
+
+  it('doesn\'t set metrics on ignored endpoint but set it on not ignored endpoint', () => {
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => {});
+    stellar.handler.get(resource2, () => {});
+
+    runMetrics({ handler: stellar.handler }, service, `${resource1}:get`);
+
+    return Promise.all([stellar.request.get(resource1), stellar.request.get(resource2)])
+        .then(() => {
+          expect(getMetrics()[`${resource1}:get`].requests).toBe(1);
+          expect(getMetrics()[`${resource2}:get`]).not.toBeDefined();
+        });
+  });
+
+  it('sets metrics to 1 after 1 request, a reset and a request', () => {
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => {});
+
+    runMetrics({ handler: stellar.handler }, service, '.*');
+
+    return stellar.request.get(resource1)
+        .then(() => {
+          expect(getMetrics()[`${resource1}:get`].requests).toBe(1);
+        })
+        .then(() => resetMetrics())
+        .then(() => stellar.request.get(resource1))
+        .then(() => {
+          expect(getMetrics()[`${resource1}:get`].requests).toBe(1);
+        });
+  });
+
+  it('gets metrics from the metrics endpoint', () => {
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => {});
+
+    runMetrics({ handler: stellar.handler }, service, '.*');
+
+    return stellar.request.get(resource1)
+        .then(() => stellar.request.get(`${service}:metrics`))
+        .then(({ metrics, node }) => {
+          console.log(node);
+          expect(metrics[`${resource1}:get`].requests).toBe(1);
+        });
+  });
+
+  it('sets failed metrics to 1 after 1 failed request', () => {
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => { throw new Error(); });
+
+    runMetrics({ handler: stellar.handler }, service, '.*');
+
+    return stellar.request.get(resource1)
+        .catch(() => {
+          expect(getMetrics()[`${resource1}:get`].failedRequests).toBe(1);
+          expect(getMetrics()[`${resource1}:get`].requests).toBe(1);
+        });
+  });
+
+  xit('gets metrics from the subscription', (done) => {
+    // TODO needs to figure out timers faking
+    const stellar = createStellar();
+    stellar.handler.get(resource1, () => {});
+
+    jest.useFakeTimers();
+    runMetrics({ handler: stellar.handler, pubSub: stellar.pubSub }, service, '.*');
```

Comment:
hack your interval value somehow and then this should work fine without fakeTimers